File: app/src/main/java/com/abs/launcher/DaemonThread.java
package com.abs.launcher;

import android.os.Handler;
import android.os.HandlerThread;

import java.lang.ref.WeakReference;

/**
 * Created by ZY on 2017/4/2.
 */

public class DaemonThread {
    private static final HandlerThread WORK_THREAD = new HandlerThread("Launcher-Daemon");
    private static final Handler HANDLER;
    static {
        WORK_THREAD.start();
        HANDLER = new Handler(WORK_THREAD.getLooper());
    }

    public static void execute(Runnable runnable) {
        HANDLER.post(runnable);
    }

    public static void stop(Runnable runnable) {
        HANDLER.removeCallbacks(runnable);
    }
}

File: app/src/main/java/com/abs/launcher/DaemonThread.kt
package com.abs.launcher

import android.os.Handler
import android.os.HandlerThread

/**
 * Created by zy on 17-12-18.
 */
val WORK_THREAD: HandlerThread = HandlerThread("launcher-daemon").apply { this.start() }
val WORK_HANDLER: Handler = Handler(WORK_THREAD.looper)

fun Runnable.runInDaemon() = WORK_HANDLER.post(this)
fun Runnable.exitDaemon() = WORK_HANDLER.removeCallbacks(this)
fun runInDaemon(f: () -> Unit) = WORK_HANDLER.post(f)
